### PR TITLE
Move dtor of BetaRowset to cpp file

### PR DESCRIPTION
**Pull request by lingbin**

### CODE REVIEW

File: be/src/olap/rowset/beta_rowset.cpp [hunk 1]
```diff
@@ -97,11 +101,13 @@ OLAPStatus BetaRowset::remove() {
 OLAPStatus BetaRowset::link_files_to(const std::string& dir, RowsetId new_rowset_id) {
     for (int i = 0; i < num_segments(); ++i) {
         std::string dst_link_path = segment_file_path(dir, new_rowset_id, i);
+        // TODO(lingbin): use Env API? or EnvUtil?
         if (FileUtils::check_exist(dst_link_path)) {
             LOG(WARNING) << "failed to create hard link, file already exist: " << dst_link_path;
             return OLAP_ERR_FILE_ALREADY_EXIST;
         }
         std::string src_file_path = segment_file_path(_rowset_path, rowset_id(), i);
+        // TODO(lingbin): how bos support link? use copy? or keep refcount to avoid being delete?
```
Comment (imay) [inline be/src/olap/rowset/beta_rowset.cpp]: What is bos? Can we change bos to external storage?

File: be/src/olap/rowset/beta_rowset.cpp [hunk 2]
```diff
@@ -60,7 +63,7 @@ OLAPStatus BetaRowset::do_load_once(bool use_cache) {
 
 OLAPStatus BetaRowset::create_reader(RowsetReaderSharedPtr* result) {
     RETURN_NOT_OK(load());
-    result->reset(new BetaRowsetReader(std::static_pointer_cast<BetaRowset>(shared_from_this())));
+    result->reset(new BetaRowsetReader(std::dynamic_pointer_cast<BetaRowset>(shared_from_this())));
```
Comment (imay) [inline be/src/olap/rowset/beta_rowset.cpp]: better to use static to better performance.

**Review by imay — APPROVED** [review]
LGTM

File: be/src/olap/rowset/beta_rowset.cpp (hunk 1 above)
Comment (lingbin) [reply be/src/olap/rowset/beta_rowset.cpp]: Done.

**Pull request merged**